plugins/modules/win_pssession_configuration.ps1

```diff
@@ -203,6 +203,155 @@ function ConvertFrom-AnsibleOption {
     }
 }
 
+function ConvertTo-PsNative {
+    <#
+    .SYNOPSIS
+    Normalize runtime objects to PowerShell-native types.
+
+    .DESCRIPTION
+    Semantics:
+    - Unwraps PSObject to BaseObject
+    - Converts IDictionary -> Hashtable (recursively) with string keys
+    - Converts IEnumerable -> array of parsed elements (recursively), preserving strings
+
+    .NOTES
+    - This is used for Ansible argument spec entries declared as "type='raw'"
+        so that module parameters arriving as CLR or PSObject-wrapped values
+        are made PowerShell-native before being passed to PowerShell cmdlets.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        $Value
+    )
+
+    End {
+        if ($null -eq $Value) { return $null }
+
+        # Unwrap PSObject wrapper if present
+        if ($Value -is [System.Management.Automation.PSObject] -and $Value.BaseObject -ne $null) {
+            $Value = $Value.BaseObject
+        }
+
+        # IDictionary -> Hashtable
+        if ($Value -is [System.Collections.IDictionary]) {
+            $ht = @{}
+            foreach ($k in $Value.Keys) {
+                $v = $Value[$k]
+                if ($k -is [string]) { $keyName = $k } else { $keyName = $k.ToString() }
+                $ht[$keyName] = ConvertTo-PsNative -Value $v
+            }
+            return $ht
+        }
+
+        # IEnumerable (but not string) -> array of parsed elements
+        if ($Value -is [System.Collections.IEnumerable] -and -not ($Value -is [string])) {
```

Comment:
I think we can just have this check for `IList` to remove the `String` check and make it simpler. I cannot think of any reason why we wouldn't have anything but an `IList` here

```powershell
if ($Value -is [System.Collections.IList]) {
}
```